### DISCUSSION TITLE
Best way to invert boolean field?

### DISCUSSION THREAD

**Discussion by mrlubos** [react-hook-form/react-hook-form#6553] [Q&A] [answered]
Hey everyone! I am struggling to find in the documentation the best practice for transforming rendered output. I have a form that outputs a boolean field `foo` which then gets passed to other functions in the application. The corresponding UI element should display the inverted value.

Example: There's a `user_opt_in_tracking` boolean value the client needs to collect before allowing the user to access the application. The UI element for this field will say "do not track me," i.e. inverse of the underlying data value.

The way my application currently works is by registering the input ID, but that maps the value directly, leading to an incorrect UI state. I could register a new field `user_do_not_track` and invert it on submit, but that's a very complicated way of handling things when all I'm trying to do is transform the UI output. I am sure there's a way to do it and I just can't find it.

Thanks for any help!

**Comment by slugmandrew** [comment, marked as answer]
This doesn't seem related to this library, however maybe you just need something like `<input type="checkbox" checked={!user_opt_in_tracking} />`?

That way your checkbox will be checked if the value is `false`, and unchecked when the value is `true`. Maybe I'm misunderstanding what you want to do? If so, try to create a codesandbox demonstrating your issue and I will try to help.

**Reply by mrlubos** [reply]
Hi @slugmandrew, yes, that's what I'm trying to do! However, I'd like to avoid using `watch('user_opt_in_tracking')` if possible as I'd need to pass it between components. The checkbox already knows its state, so I wonder if it can be exposed and modified. That way should be more performant too as I'll be modifying only that checkbox instead of re-rendering my whole component on `watch()` change.

**Reply by slugmandrew** [reply]
I'm really not sure what you are trying to do. You need to provide more information or a proper example.

I'd highly recommend taking the time to create one, even if it takes a few hours, it's better than the hours or days you'll waste trying to _describe_ your problem when you could just *show* me.

**Reply by mrlubos** [reply]
Thanks @slugmandrew. I've been digging in the issues and API and found a way to achieve what I want to do via `useController()` interface which exposes `field` object with the `value` key.

**Reply by bluebill1049** [reply]
thanks @slugmandrew for helping 🙏

**Reply by slugmandrew** [reply]
@mrlubos @bluebill1049 Glad I could help!

**Reply by bluebill1049** [reply]
i have include your name: https://github.com/react-hook-form/react-hook-form/discussions/5340 🙏

**Reply by mrlubos** [reply]
Ah, turns out in the end the `field` approach didn't work. While I was able to manipulate the rendered value (in my case by doing `value={!field.value}`), I ran into another issue with `onChange()` handler. I have found another solution not involving react-hook-form, but if I missed some feature please let me know!

To summarise, the feature I am looking for is for formatting rendered output, not the underlying data. You can imagine a field for printing value in degrees Fahrenheit or Celsius, but the underlying data from API would always come in Celsius. Ideally, I wouldn't have to do back-and-forth conversions when all I am really after is render `data.value` when in Celsius mode and `(data.value * 1.8) + 32` when in Fahrenheit mode.
